ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLAgentExecutor.java

```diff
@@ -124,28 +132,27 @@ public void execute(Input input, ActionListener<Output> listener) {
                                 conversationIndexMemoryFactory.create(question, memoryId, appType, ActionListener.wrap(memory -> {
                                     inputDataSet.getParameters().put(MEMORY_ID, memory.getConversationId());
-                                    // Create root interaction ID
-                                    ConversationIndexMessage msg = ConversationIndexMessage
-                                        .conversationIndexMessageBuilder()
-                                        .type(appType)
-                                        .question(question)
-                                        .response("")
-                                        .finalAnswer(true)
-                                        .sessionId(memory.getConversationId())
-                                        .build();
-                                    memory.save(msg, null, null, null, ActionListener.<CreateInteractionResponse>wrap(interaction -> {
-                                        log.info("Created parent interaction ID: " + interaction.getId());
-                                        inputDataSet.getParameters().put(PARENT_INTERACTION_ID, interaction.getId());
-                                        ActionListener<Object> agentActionListener = createAgentActionListener(
-                                            listener,
-                                            outputs,
-                                            modelTensors
-                                        );
-                                        executeAgent(inputDataSet, mlAgent, agentActionListener);
-                                    }, ex -> {
-                                        log.error("Failed to create parent interaction", ex);
-                                        listener.onFailure(ex);
-                                    }));
+                                    ActionListener<Object> agentActionListener = createAgentActionListener(listener, outputs, modelTensors);
+                                    // get question for regenerate
+                                    if (regenerateInteractionId != null) {
+                                        log.info("Regenerate for existing interaction {}", regenerateInteractionId);
+                                        client
+                                            .execute(
+                                                GetInteractionAction.INSTANCE,
+                                                new GetInteractionRequest(memoryId, regenerateInteractionId),
+                                                ActionListener.wrap(interactionRes -> {
+                                                    inputDataSet
+                                                        .getParameters()
+                                                        .computeIfAbsent(QUESTION, (key) -> interactionRes.getInteraction().getInput());
```

Comment:
If the original question had verbose enabled, does it need to be honored for regenerate request?

Reply:
I don't think we stored the request detail like `verbose` parameter, and from application perspective, each request have same parameters.

Reply:
For current chat implementation, we don't have UI for user to edit original question. we want API to support this when enabling this in future.